docs/admonitions.rst

```diff
@@ -2,219 +2,318 @@
 Admonitions
 ===========
 
-This theme uses the :dutree:`admonition` directives for Sphinx admonitions.
+This theme uses the `Admonition Directive`_ for Sphinx admonitions.
+
+.. _predefined_admonitions:
 
 rST and Sphinx
 **************
 
 Most of the admonitions that the mkdocs-material theme supports were "borrowed" from
 admonitions defined in the reStructuredText specifications. You may recognize them from
-usage in other sphinx-based themes. They are:
+usage in other sphinx-based themes.
 
-.. rst-example:: ``note``, ``todo``, ``seealso``
+.. rst-example:: ``note``, ``seealso``, ``todo``
+   :name: note-style
 
    .. seealso::
       This admonition is specific to Sphinx directives and not defined in the reStructuredText
-      specifications as you can `seealso`. The `todo` admonition is also defined by Sphinx.
+      specifications as you can `seealso`. The `todo` admonition is also defined as an extension
+      that is distributed with Sphinx.
 
       The :dutree:`note` admonition *is* defined by the reStructuredText specifications.
 
 .. rst-example:: ``tip``, ``hint``, ``important``
+   :name: important-style
 
    .. important::
       It is :dutree:`important` to correctly use admonitions.
 
 .. rst-example:: ``attention``, ``caution``, ``warning``
+   :name: attention-style caution-style warning-style
 
    .. warning::
       This is a :dutree:`warning`.
 
 .. rst-example:: ``danger``, ``error``
+   :name: error-style
 
    .. error::
       You have made a grave :dutree:`error`.
 
+.. _inherited_admonitions:
+
 Admonitions from mkdocs-material
 ********************************
 
-Some additional admonitions are supported via the source code from the mkdocs-material theme.
-These admonitions can still be used, but the syntax is a little different because it relies
-on the generic :dutree:`admonition` defined in the reStructuredText specifications.
+Some additional admonitions are supported via the CSS styles inherited from the mkdocs-material
+theme. These admonition styles can be used via the :rst:dir:`admonition` directive's :rst:`:class:`
+option. 
+
+.. confval:: sphinx_immaterial_generate_inherited_admonitions
```

Comment:
I think originally you just defined an `xxx` directive for each `xxx` admonition class.  I was concerned about polluting the directive namespace.  However, I expect in practice users will want to use these directives with nice shorthand names, rather than `si-xxx`.  Therefore, it might be better for this option to just control whether we provide the `xxx` directives --- potentially we could always provide the `si-xxx` directives since they shouldn't cause any conflicts.

However, it might also be convenient to provide a way for a user to easily define their own admonition directives that map to the built-in admonition styles.  Perhaps instead of this option, we could provide a way for the user to specify these admonitions via `custom_admonitions`, e.g. with an option to have the admonition directive just map to an existing class rather than define a new class.

Reply:
>  Therefore, it might be better for this option to just control whether we provide the xxx directives

I'll revert the admonition name changes, so the option will be more useful about polluting directives' names.

>  Perhaps instead of this option, we could provide a way for the user to specify these admonitions via `custom_admonitions`, e.g. with an option to have the admonition directive just map to an existing class rather than define a new class.

I too was wondering how we could go about this. I hadn't considered adding a config option though, that would make things simpler. I think a `default_class` option would satisfy this well.

Reply:
Done. I chose the name `classes` to accept a list of CSS class names. 
```py
sphinx_immaterial_custom_admonitions = [
    {
        "name": "my-note",
        "classes": ["note"],
    }
]
```
Although I'm not sure how useful it would be to specify multiple class names for a single directive...